core/src/main/scala/akka/kafka/scaladsl/MetadataClient.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka.scaladsl
+
+import akka.actor.ActorRef
+import akka.kafka.Metadata.{BeginningOffsets, GetBeginningOffsets}
+import akka.pattern.ask
+import akka.util.Timeout
+import org.apache.kafka.common.TopicPartition
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object MetadataClient {
+
+  def getBeginningOffsets(
+      consumerActor: ActorRef,
+      partitions: Set[TopicPartition],
+      timeout: Timeout
+  )(implicit ec: ExecutionContext): Future[Map[TopicPartition, Long]] =
+    (consumerActor ? GetBeginningOffsets(partitions))(timeout)
+      .mapTo[BeginningOffsets]
+      .map(_.response.get)
```

Comment:
This will throw an exception if there was an error and fail the future that way. Instead, you should `transform` the `Future`. To be Scala 2.11 compatible you need to do it like this 
```
    .flatMap {
      case Success(res) => Future.successful(res)
      case Failure(e) => Future.failed(e)
    }(ExecutionContexts.sameThreadExecutionContext)
```
With this it is OK to use Akka's `sameThreadExecutionContext` to run the `flatMap` on.

Reply:
Do we really need this? I added failing test cases and the code behaves the same for:
```
.map(_.response.get)
```
as well as for:
```
.map(_.response)
      .flatMap {
        case Success(res) => Future.successful(res)
        case Failure(exception) => Future.failed(exception)
      }
```

Reply:
To run it safely on the `sameThreadExecutionContext` we may not allow any exceptions happening in the operator.